### PR TITLE
Support setting Server GC and other properties from runtimeconfig.json via MSBuild properties

### LINKED ISSUE #474: Support setting Server GC and other properties from runtimeconfig.json via MSBuild properties
Currently, to use the Server GC, you need to have a `runtimeconfig.template.json` file with the setting, which is then [merged](https://github.com/dotnet/sdk/blob/34aa24bba2829a81719eb8c1c901db52df508704/src/Tasks/Microsoft.NET.Build.Tasks/GenerateRuntimeConfigurationFiles.cs) into the runtimeconfig.json that the app uses.

We should add an MSBuild property to set the GC mode (and any other important properties from the runtimeconfig) that will get written to the output runtimeconfig without having to have the template .json file in your project.  The Web Sdk should set this property to Server GC by default.

### PR DESCRIPTION
**Pull request by eerhardt**
@nguerrera @dsplaisted @srivatsn 

/cc @davidfowl @mlorbetske @DamianEdwards 

With this change the Web.Sdk can default `<ServerGarbageCollection>true</ServerGarbageCollection>` in their .props file, and server GC will be turned on for .NET Core apps.

Fix #474

### CODE REVIEW

**Comment by nguerrera** [comment]
I don't think we'll get this past ship room so it'll probably have to go in future.

**Review by nguerrera — APPROVED** [review]
I like the feature and the code but it doesn't meet the RTW bar IMHO. 

Note that you need to fill out the template (see other recent PRs) and tag Matt Gertz for approval to bring to ship room then wait for approval to merge.

If it gets rejected, ship it in future.

**Comment by eerhardt** [comment]
@davidfowl @mlorbetske @DamianEdwards - if you guys want me to push this for "1.0", let me know.  But if you don't want/need it, I can just push it to "future".

**Comment by DamianEdwards** [comment]
@eerhardt I'd like us to push for this in 1.0, thanks.

**Comment by eerhardt** [comment]
@MattGertz for approval

Scenario

ASP.NET projects want Server Garbage Collection turned on by default.  The only way to do this today is to add a new file `runtimeconfig.template.json` in the project and set a single property in the JSON.  This clutters up the ASP.NET project with yet another .json file (there are already 5 .json files in the root of the project for a "StarterWeb" template).

Instead, this allows the setting to be controlled by an MSBuild property. The Web.Sdk could default "Server Garbage Collection" to on without affecting the user's project or templates at all.  Or the ASP.NET team can add the single MSBuild property into the web template.  However, they choose to take advantage of this feature.

Either way, the developer still has full control, and can turn the property back off, if they don't want server garbage collection.

Bug

#474

Workarounds

The workaround is to add a runtimeconfig.template.json file to the project that looks like:

```JSON
{
  "configProperties": {
    "System.GC.Concurrent": true
  }
} 
```

Risk

Low - new functionality with no change to existing functionality

Performance Impact

For build time, no impact. 
For runtime, using server garbage collection in web apps can improve the performance of the web application. See https://msdn.microsoft.com/en-us/library/ee787088(v=vs.110).aspx#workstation_and_server_garbage_collection

Regression Analysis

Not a regression

**Comment by dasMulli** [comment]
Does this only modify the `runtimeconfig.json` or will this also update/generate any present `App.config` / `web.config` for full framework applications?

**Comment by eerhardt** [comment]
> Does this only modify the runtimeconfig.json or will this also update/generate any present App.config / web.config for full framework applications?

This is only for .NET Core applications.  Full framework applications can still set the property in the App.config/Web.config.  A future enhancement could take these properties and set the values in the app/web.config.

**Comment by MattGertz** [comment]
Sri/JoC/I are discussing.

**Comment by MattGertz** [comment]
Talked to JoC; OK to bring to Shiproom.  Cost/risk will be the discussion there.

**Comment by eerhardt** [comment]
@

**Comment by eerhardt** [comment]
@MattGertz - I don't have an invite for shiproom.  Let me know if there is anything for me to do, or send me the invite if you want me to attend.

**Comment by MattGertz** [comment]
No worries, Sri usually speaks for SDK, and John's already aware of the existence of this one anyway now.

**Comment by natidea** [comment]
Approved to merge via [377254](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/377254) and pending one more review

**Review by srivatsn — APPROVED** [review]

**Comment by srivatsn** [comment]
Thanks Eric!

**Pull request merged**
